### PR TITLE
 [zuul] Update the external_plugin var

### PR DESCRIPTION
**Pull request by mgirgisf**

### CODE REVIEW

**Comment by elfiesmelfie** [comment]
This won't work until the telemetry-tempest-plugin is tagged and released. Is that the intention?

**Comment by mgirgisf** [comment]
yes exactly, waiting until telemetry-tempest-plugin is tagged and released

**Comment by elfiesmelfie** [comment]
This'll hopefully start working soon, the release patch was proposed and approved: https://review.opendev.org/c/openstack/releases/+/913549

**Comment by elfiesmelfie** [comment]
I've update this PR, since the new package has propagated to the tempest container image

File: ci/vars-autoscaling-tempest.yml [hunk 1]
```diff
@@ -17,3 +17,7 @@ cifmw_test_operator_tempest_include_list: |
   telemetry_tempest_plugin.scenario
   telemetry_tempest_plugin.aodh
 cifmw_test_operator_tempest_exclude_list: ''
+external_plugin: "opendev.org/openstack/telemetry-tempest-plugin"
+change_item: "{{ zuul['items'] | selectattr('project.canonical_name', 'equalto', external_plugin) }}"
+# This value will be used when the telemetry-tempest-plugin is released, and we no longer need to install from the master branch
+cifmw_test_operator_tempest_external_plugin: "{{ [] if change_item | length < 1 else [ { 'repository': 'https://' + external_plugin + '.git', 'changeRepository': 'https://review' + {{ external_plugin }}, 'changeRefspec': [ 'refs/changes', change_item[0].change[-2:], change_item[0].change, change_item[0].patchset ] | join('/') } ] }}"
```
Comment (elfiesmelfie) [inline ci/vars-autoscaling-tempest.yml:23]:
```suggestion
cifmw_test_operator_tempest_external_plugin: "{{ [] if change_item | length < 1 else [ { 'repository': 'https://' + external_plugin + '.git', 'changeRepository': 'https://review' + external_plugin, 'changeRefspec': [ 'refs/changes', change_item[0].change[-2:], change_item[0].change, change_item[0].patchset ] | join('/') } ] }}"
```

**Comment by mgirgisf** [comment]
recheck

**Comment by elfiesmelfie** [comment]
> Build failed (check pipeline). Post `recheck` (without leading slash) to rerun all jobs. Make sure the failure cause has been resolved before you rerun jobs.
> 
> https://review.rdoproject.org/zuul/buildset/7d697651719a4dd395414052d66df518
> 
> ✔️ [openstack-k8s-operators-content-provider ](https://review.rdoproject.org/zuul/build/dc5cd6a6b4b9475b90724d0a82f5732d) SUCCESS in 1h 46m 58s ✔️ [podified-multinode-edpm-deployment-crc ](https://review.rdoproject.org/zuul/build/bfd8447c7a614185aa45272896618160) SUCCESS in 1h 11m 28s ❌ [telemetry-operator-multinode-autoscaling-tempest ](https://review.rdoproject.org/zuul/build/1ade40b4fd5242728fbdbe1b418da7e4) **FAILURE** in 1h 11m 47s

This failure is a tempest failure!
https://logserver.rdoproject.org/23/323/09df2e335b780e51c450026a97c38e87f72c8ea3/github-check/telemetry-operator-multinode-autoscaling-tempest/1ade40b/controller/ci-framework-data/tests/test_operator/tempest-tests/stestr_results.html

Looks like a URL needs to be updated

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>400 Bad Request</title>
</head><body>
<h1>Bad Request</h1>
<p>Your browser sent a request that this server could not understand.<br />
Reason: You're speaking plain HTTP to an SSL-enabled server port.<br />
 Instead use the HTTPS scheme to access this URL, please.<br />
</p>
</body></html>
```

**Review by elfiesmelfie — APPROVED** [review]

File: ci/vars-autoscaling-tempest.yml [hunk 2]
```diff
@@ -12,8 +12,12 @@ cifmw_tempest_tempestconf_config:
       service_available.ceilometer true
       service_available.sg_core true
       service_available.aodh true
-      telemetry.sg_core_service_url "ceilometer-internal.openstack.svc.cluster.local:3000"
+      telemetry.sg_core_service_url "https://ceilometer-internal.openstack.svc.cluster.local:3000"
```
Comment (csibbitt) [inline ci/vars-autoscaling-tempest.yml:15]: https://github.com/openstack-k8s-operators/telemetry-operator/pull/369/files#r1570893833

File: ci/vars-autoscaling-tempest.yml [hunk 3]
```diff
@@ -12,8 +12,12 @@ cifmw_tempest_tempestconf_config:
...
 cifmw_test_operator_tempest_include_list: |
   telemetry_tempest_plugin.scenario
   telemetry_tempest_plugin.aodh
 cifmw_test_operator_tempest_exclude_list: ''
+external_plugin: "opendev.org/openstack/telemetry-tempest-plugin"
```
Comment (tosky) [inline ci/vars-autoscaling-tempest.yml:20]: Is this for testing, or is a new release of the plugin coming?

Comment (elfiesmelfie) [reply ci/vars-autoscaling-tempest.yml:20]: This vars is defined for convenience.  It is used when defining the cifmw_test_operator_tempest_external_plugin var.
The cifmw_test_operator_tempest_external_plugin var will normally get set to '[]' (it's default) but when there's a depends-on, it'll be configured to use the change from opendev/telemetry-tempest-plugin.

The change in this pr is switching from configuring cifmw_test_operator_tempest_external_plugin to use master to resetting it to default.
The depends-on piece is so we can test wip patches to telemetry-tempest-plugin. 
This is the basis for a job I'm working on to allow rdo zuul to run this same job on changes to opendev so we can know that the tempest upstream tests are still working for us, and for adding additional autoscaling tempest tests, and knowing they'll continue to work across versions

File: ci/vars-autoscaling-tempest.yml [hunk 4]
```diff
@@ -12,8 +12,12 @@ cifmw_tempest_tempestconf_config:
...
+change_item: "{{ zuul['items'] | selectattr('project.canonical_name', 'equalto', external_plugin) }}"
+# This value will be used when the telemetry-tempest-plugin is released, and we no longer need to install from the master branch
```
Comment (elfiesmelfie) [inline ci/vars-autoscaling-tempest.yml:22]: This comment should be removed. I added it in when I first tested the template for setting cifmw_test_operator_tempest_external_plugin based on depends-on.

File: ci/vars-autoscaling-tempest.yml (hunk 3 above)
Comment (elfiesmelfie) [reply ci/vars-autoscaling-tempest.yml:20]: You can see the previous values in zuul.d/project.yaml.

Comment (elfiesmelfie) [reply ci/vars-autoscaling-tempest.yml:20]: Also, setting the `external_plugin`  makes it easier for other operator jobs to use this pattern for testing patches from opendev.

File: ci/vars-autoscaling-tempest.yml (hunk 4 above)
Comment (elfiesmelfie) [inline ci/vars-autoscaling-tempest.yml:22]:
```suggestion
```

**Pull request merged**